spec/features/profiles/show_spec.rb

```diff
@@ -17,8 +17,28 @@
       visit "/profiles/#{profile.id}".dup
     end
 
-    it "shows the PROFILE" do
+    it "shows the profile" do
       expect(page).to have_content "ChaelCodes"
+      expect(page).not_to have_link "Edit", href: edit_profile_path(profile)
+      expect(page).not_to have_link "Delete", href: profile_path(profile)
+    end
+
+    context "when profile belongs to the user" do
+      let(:user) { profile.user }
+
+      it "allows user to edit profile" do
+        expect(page).to have_link "Edit", href: edit_profile_path(profile)
+        expect(page).to have_link "Delete", href: profile_path(profile)
+      end
+    end
+
+    context "when user is admin" do
+      let(:user) { create :user, :admin }
+
+      it "allows user to destroy profile" do
```

Comment:
(same)
Is this actually testing that the user can destroy the event, or rather just that the destroy link shows up on the page. I think you have the "can destroy the event" is authorization in the request spec.